backend/tests/daily/connectors/google_drive/test_google_drive_slim_docs.py

```diff
@@ -0,0 +1,156 @@
+import time
+from collections.abc import Callable
+from unittest.mock import MagicMock
+from unittest.mock import patch
+
+from danswer.access.models import ExternalAccess
+from danswer.connectors.google_drive.connector import GoogleDriveConnector
+from danswer.connectors.google_drive.google_utils import execute_paginated_retrieval
+from ee.danswer.external_permissions.google_drive.doc_sync import (
+    _get_permissions_from_slim_doc,
+)
+from tests.daily.connectors.google_drive.helpers import DRIVE_MAPPING
+from tests.daily.connectors.google_drive.helpers import flatten_file_ranges
+from tests.daily.connectors.google_drive.helpers import (
+    get_expected_file_names_and_texts,
+)
+from tests.daily.connectors.google_drive.helpers import PUBLIC_RANGE
+
+
+def get_keys_available_to_user_from_access_map(
+    user_email: str,
+    group_map: dict[str, list[str]],
+    access_map: dict[str, ExternalAccess],
+) -> list[str]:
+    """
+    Extracts the names of the files available to the user from the access map
+    through their own email or group memberships or public access
+    """
+    group_emails_for_user = []
+    for group_email, user_in_group_email_list in group_map.items():
+        if user_email in user_in_group_email_list:
+            group_emails_for_user.append(group_email)
+
+    accessible_file_names_for_user = []
+    for file_name, external_access in access_map.items():
+        if external_access.is_public:
+            accessible_file_names_for_user.append(file_name)
+        elif user_email in external_access.external_user_emails:
+            accessible_file_names_for_user.append(file_name)
+        elif any(
+            group_email in external_access.external_user_group_ids
+            for group_email in group_emails_for_user
+        ):
+            accessible_file_names_for_user.append(file_name)
+    return accessible_file_names_for_user
+
+
+def check_access_for_user(
```

Comment:
for future reference, would prefer if these tests were a bit less abstracted. imo It's pretty hard to tell what's going on here (and throughout the test)

Reply:
let's chat through this (maybe as a team). but not worth re-writing these ones